pkg/cover/instrument.go

```diff
@@ -116,6 +122,54 @@ func loadFileCover(coverCounters map[string][]uint32, coverBlocks map[string][]t
 	coverBlocks[fileName] = block
 }
 
+func clearValues() (map[string][]uint32, map[string][]testing.CoverBlock) {
+	var (
+		coverCounters = make(map[string][]uint32)
+		coverBlocks   = make(map[string][]testing.CoverBlock)
+	)
+
+	lock.Lock()
+	defer lock.Unlock()
+
+	{{range $i, $pkgCover := .DepsCover}}
+	{{range $file, $cover := $pkgCover.Vars}}
+	clearFileCover(coverCounters, coverBlocks, {{printf "%q" $cover.File}}, _cover{{$i}}.{{$cover.Var}}.Count[:], _cover{{$i}}.{{$cover.Var}}.Pos[:], _cover{{$i}}.{{$cover.Var}}.NumStmt[:])
+	{{end}}
+	{{end}}
+
+	{{range $file, $cover := .MainPkgCover.Vars}}
+	clearFileCover(coverCounters, coverBlocks, {{printf "%q" $cover.File}}, {{$cover.Var}}.Count[:], {{$cover.Var}}.Pos[:], {{$cover.Var}}.NumStmt[:])
+	{{end}}
+
+	{{range $k, $pkgCover := .CacheCover}}
+	{{range $v, $cover := $pkgCover.Vars}}
+	clearFileCover(coverCounters, coverBlocks, {{printf "%q" $cover.File}}, {{$pkgCover.Package.Name}}.{{$v}}.Count[:], {{$pkgCover.Package.Name}}.{{$v}}.Pos[:], {{$pkgCover.Package.Name}}.{{$v}}.NumStmt[:])
+	{{end}}
+	{{end}}
+
+	return coverCounters, coverBlocks
+}
+
+func clearFileCover(coverCounters map[string][]uint32, coverBlocks map[string][]testing.CoverBlock, fileName string, counter []uint32, pos []uint32, numStmts []uint16) {
```

Comment:
the `counter []uint32` seems to be a value type，i am curious why this method `clearFileCover ` can clear the internal struct state of the services under test...